### PR TITLE
Add support for Package.resolved v2

### PR DESCRIPTION
**Pull request by Kyle-Ye**
- Add Package.resolved v2 decoding support
- Align PackageResolvedV1 definition with [swift-package-manager](https://github.com/apple/swift-package-manager)

### CODE REVIEW

**Comment by Kyle-Ye** [comment]
See more info here cc @0xTim 

https://github.com/apple/swift-package-manager/blob/f04ad469a6053d713c2fb854fbeb27ee3e6c9dee/Sources/PackageGraph/PinsStore.swift#L225-L403

File: Sources/VaporToolbox/Toolbox.swift [hunk 1]
```diff
@@ -52,20 +52,46 @@ final class Toolbox: CommandGroup {
...
-            if let vapor = package.object.pins.first(where: { $0.package == "vapor" }) {
-                if let version = vapor.state.version {
-                    context.console.output(key: "framework", value: version)
-                } else if let branch = vapor.state.branch {
-                    context.console.output(key: "framework", value: "Branch-\(branch) Unknown version")
-                } else  {
-                    context.console.output(key: "framework", value: "Revision-\(vapor.state.revision) Unknown version")
-                }
-            } else {
+            func missingVaporOutput(){
```
Comment (0xTim) [inline Sources/VaporToolbox/Toolbox.swift:55]: Nit formatting
```suggestion
            func missingVaporOutput() {
```

File: Sources/VaporToolbox/Toolbox.swift [hunk 2]
```diff
@@ -52,20 +52,46 @@ final class Toolbox: CommandGroup {
...
+                    } else if let revision = vapor.state.revision {
+                        context.console.output(key: "framework", value: "Revision-\(revision) Unknown version")
+                    } else {
+                        context.console.output(key: "framework", value: "Unknown version")
+                    }
+                } else {
+                    missingVaporOutput()
+                }
+            default:
+                context.console.output(key: "framework", value: "Not supported Package.resolved version")
```
Comment (0xTim) [inline Sources/VaporToolbox/Toolbox.swift:93]: Nit grammar
```suggestion
                context.console.output(key: "framework", value: "Unsupported Package.resolved version")
```

File: Sources/VaporToolbox/Toolbox.swift [hunk 3]
```diff
@@ -96,22 +122,46 @@ final class Toolbox: CommandGroup {
     }
 }
 
-private struct PackageResolved: Codable {
+private struct Version: Codable {
+    let version: Int
+}
+    
+
+
```
Comment (0xTim) [inline Sources/VaporToolbox/Toolbox.swift:130]: Nit extra space
```suggestion
    
```

**Review by 0xTim — CHANGES_REQUESTED** [review]
Thanks! Few nits but mostly there

**Review by 0xTim — APPROVED** [review]

**Pull request merged**